File: evals/fixtures/react-native-expo-mobile/RN-ASYNCSTORAGE-001/files/authStorage.ts
import AsyncStorage from "@react-native-async-storage/async-storage";

// Vulnerable: AsyncStorage on Android backs onto an unencrypted SQLite
// database and on iOS onto unencrypted plist/SQLite files - neither is
// designed for secrets. A long-lived refresh token stored here is
// recoverable from a device backup, from an Android app with a targeted
// backup exploit, or trivially on a rooted/jailbroken device. This should
// use expo-secure-store (Keychain/Keystore-backed) instead.
export async function saveRefreshToken(token: string): Promise<void> {
  await AsyncStorage.setItem("refresh_token", token);
}

export async function getRefreshToken(): Promise<string | null> {
  return AsyncStorage.getItem("refresh_token");
}

File: evals/fixtures/flutter-dart-mobile/FLUTTER-INSECURESTORE-001/files/lib/auth_storage.dart
import 'package:shared_preferences/shared_preferences.dart';

/// Vulnerable: the long-lived refresh token is written to SharedPreferences,
/// which on Android backs onto an unencrypted XML file and on iOS onto an
/// unencrypted plist - both readable without root/jailbreak on a device
/// with a file-system backup enabled, or trivially on a rooted/jailbroken
/// device. This should use flutter_secure_storage (Keystore/Keychain)
/// instead, which is designed for exactly this kind of long-lived secret.
class AuthStorage {
  Future<void> saveRefreshToken(String token) async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString('refresh_token', token);
  }

  Future<String?> getRefreshToken() async {
    final prefs = await SharedPreferences.getInstance();
    return prefs.getString('refresh_token');
  }
}
